server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/environments_widget.tsx

```diff
@@ -54,27 +55,45 @@ export class EnvironmentWidget extends MithrilViewComponent<EnvAttrs> {
 
   view(vnode: m.Vnode<EnvAttrs>) {
     const environment = vnode.attrs.environment;
-    const isEnvEmpty  = _.isEmpty(environment.pipelines()) && _.isEmpty(environment.agents());
     return <Anchor id={environment.name()} sm={vnode.attrs.sm} onnavigate={() => this.expanded(true)}>
       <CollapsiblePanel header={<EnvironmentHeader environment={environment}/>}
-                        warning={isEnvEmpty}
-                        actions={[
-                          <IconGroup>
-                            <Delete
-                              title={environment.canAdminister() ? undefined : `You dont have permissions to delete '${environment.name()}' environment.`}
-                              disabled={!environment.canAdminister()}
-                              onclick={vnode.attrs.onDelete.bind(vnode.attrs, environment)}/>
-                          </IconGroup>
-                        ]}
+                        warning={this.isEnvEmpty(environment)}
+                        actions={this.getActionButtons(environment, vnode)}
                         dataTestId={`collapsible-panel-for-env-${environment.name()}`}
                         vm={this}>
         <EnvironmentBody environment={environment}
-                         environments={vnode.attrs.environments()}
+                         environments={vnode.attrs.environments}
                          agents={vnode.attrs.agents}
                          onSuccessfulSave={vnode.attrs.onSuccessfulSave}/>
       </CollapsiblePanel>
     </Anchor>;
   }
+
+  getActionButtons(environment: EnvironmentWithOrigin, vnode: m.Vnode<EnvAttrs>) {
+    let warningButton;
+    if (this.isEnvEmpty(environment)) {
+      warningButton = <div data-test-id="warning-tooltip-wrapper" className={styles.warningTooltipWrapper}>
+        <i data-test-id={"warning-icon"} className={styles.warningIcon}/>
+        <div data-test-id="warning-tooltip-content" className={styles.warningTooltipContent}>
+          <p>Neither pipelines nor agents are associated with this environment.</p>
+        </div>
+      </div>;
+    }
+    return [
+      warningButton,
+      <IconGroup>
+        <Delete
+          title={environment.canAdminister() ? undefined : `You dont have permissions to delete '${environment.name()}' environment.`}
```

Comment:
The message can be: `You are not authorized to delete the '${environment.name()}' environment.` WDYT?